src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -0,0 +1,34 @@
+@echo off
+setlocal
+
+set PROJECT_DIR=%~1
+set DOTNET_CMD=%~2
+set TOOLRUNTIME_DIR=%~3
+set TOOLS_DIR=%~dp0
+
+if not exist "%PROJECT_DIR%" (
+  echo ERROR: Cannot find project root path at [%PROJECT_DIR%]. Please pass in the source directory as the 1st parameter.
+  exit /b 1
+)
+
+if not exist "%TOOLS_DIR%" (
+  echo ERROR: Cannot find tools path at [%TOOLS_DIR%]. Please pass in the tools directory as the 2nd parameter.
+  exit /b 1
+)
+
+if not exist "%DOTNET_CMD%" (
+  echo ERROR: Cannot find dotnet command at [%DOTNET_CMD%]. Please pass in the dotnet command as the 3rd parameter.
+  exit /b 1
+)
+
+set tempDirectory=%cd%
+cd %TOOLS_DIR%\tool-runtime\
+call %DOTNET_CMD% restore
+call %DOTNET_CMD% publish -f dnxcore50 -r win7-x64 -o %TOOLRUNTIME_DIR%
+cd %tempDirectory%
+
+xcopy /E /Y /Q %TOOLS_DIR%..\lib %TOOLRUNTIME_DIR%
+
+call type NUL > %TOOLRUNTIME_DIR%\ToolRuntime.semaphore
```

Comment:
We can do that. This semaphore file is the one used by the msbuild target to ensure that buildtools was restored, while the project.json that contains the version is used by init-tools as a semaphore to tell if we need to restore a newer version of buildtools. I'll change the logic so both msbuild and init-tools will use the project.json file as the semaphore file